File: examples/groups/SpecialGroups.rb
text = 'supermercado superação hiperMERCADO'

puts text.scan(/(?:super)[\wÀ-ú]+/i).join(' ')

# Positive Lookbehind: Matches characters behind of expected match.
puts text.scan(/(?<=super)[\wÀ-ú]+/i).join(' ')

# Negative Lookbehind: Matches characters not behind of expected match.
puts text.scan(/(?<!super)mercado/i).join(' ')

File: examples/groups/SpecialGroups.js
const text = 'João é calmo, mas no transito fica nervoso.'

const matchAllWords = /[\wÀ-ú]+/g
console.log(text.match(matchAllWords))

// Positive Lookahead: Matches characters in front of expected match.
const matchWordRightBeforeComma = /[\wÀ-ú]+(?=,)/gi
console.log(text.match(matchWordRightBeforeComma))

// Negative Lookahead: Matches characters not in front of expected match.
const matchWordWithNoMetaCharacterInFront = /[\wÀ-ú]+\b(?!,)/gi
console.log(text.match(matchWordWithNoMetaCharacterInFront))
